pkg/grpc/proxy/handler_test.go

```diff
@@ -72,9 +82,28 @@ func (s *assertingService) PingEmpty(ctx context.Context, _ *pb.Empty) (*pb.Ping
 }
 
 func (s *assertingService) Ping(ctx context.Context, ping *pb.PingRequest) (*pb.PingResponse, error) {
+	count := s.pingCallCount.Add(1)
+
+	// Simulate failures
+	if delay := s.simulateDelay.Load(); delay > 0 {
+		time.Sleep(time.Duration(delay) * time.Millisecond)
+	}
+	if s.simulatePingFailures.Add(-1) >= 0 {
+		return nil, status.Errorf(codes.Internal, "Simulated failure")
+	} else {
+		s.simulatePingFailures.Store(0)
+	}
+	if s.simulateRandomFailures.Load() {
+		if count%14 == 0 {
+			time.Sleep(800 * time.Millisecond)
+		} else if count%12 == 0 {
+			return nil, status.Errorf(codes.Internal, "Simulated random failure")
+		}
+	}
+
 	// Send user trailers and headers.
-	grpc.SendHeader(ctx, metadata.Pairs(serverHeaderMdKey, "I like turtles."))
-	grpc.SetTrailer(ctx, metadata.Pairs(serverTrailerMdKey, "I like ending turtles."))
```

Comment:
Changed the message because "ending turtles" sounds really bad

Reply:
The animal lover in me appreciates that.